File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int i=0;
        int j=heights.length-1;
        int ans = 0;

        while(i < j){
            ans = Math.max(ans, Math.min(heights[i], heights[j]) * (j-i));

            if(heights[i] < heights[j])
                i++;
            else
                j--;
        }

        return ans;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int ans = 0;
        int n = height.size();
        if(n == 1)
            return 0;

        int leftmax = height[0], i = 0, rightmax = height[n-1], j = n-1;
        
        while(i<j){
            if(leftmax <= rightmax){
                ans += leftmax - height[i];
                i++;
                leftmax = max(leftmax, height[i]);
            }
            else{
                ans += rightmax - height[j];
                j--;
                rightmax = max(rightmax, height[j]);
            }
        }
        return ans;

    }
};
